### PR TITLE
improved cancellation of monitorChannel() goroutine

### PR DESCRIPTION
**Pull request by dwhutchison**
Force cancellation of the monitorChannel() goroutine when the session is closed.  monitorChannel() currently relies on the Receive returning an error to exit but if if that doesn't occur for any reason, monitorChannel() may remain executing.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -102,6 +102,9 @@ type Client struct {
 	subscriptions map[uint32]*Subscription
 	subMux        sync.RWMutex
 
+	//cancelMonitor cancels the monitorChannel goroutine
+	cancelMonitor context.CancelFunc
```
Comment (kung-foo) [inline client.go:106]: it is normal to store a context related item in a struct?

Comment (dwhutchison) [reply client.go:106]: I hesitated with this as well but the alternative would likely be a done channel.  Using a done channel, the select statement grows by one case statement and has the same body as the ctx.Done() case.  This felt redundant so using the context seemed more natural as we were already using it. 

That said, I'm not opposed to changing it if there are strong arguments against it.

Comment (magiconair) [reply client.go:106]: I think this is OK. If the context would need to cancel more than just the monitor I'd suggest to make it more generic but this is good enough.

**Comment by magiconair** [comment]
I'll merge and tag a new version.

**Pull request merged**